### ISSUE TITLE
Plugin crashes when pushing a branch delete

### ISSUE THREAD

**Issue by itay** [christiangalsterer/stash-filehooks-plugin#2]
When you push a branch delete the plugin crashes with:

```
java.lang.IllegalStateException: At least one changeset ID must be provided to retrieve detailed commits
        at com.atlassian.stash.content.DetailedChangesetsRequest$Builder.buildChangesetIds(DetailedChangesetsRequest.java:119) ~[stash-api-3.6.0.jar:na]
        at com.atlassian.stash.content.DetailedChangesetsRequest$Builder.build(DetailedChangesetsRequest.java:83) ~[stash-api-3.6.0.jar:na]
        at org.christiangalsterer.stash.filehooks.plugin.hook.FileSizeHook$7.get(FileSizeHook.java:146) ~[na:na]
        at com.atlassian.stash.util.PagedIterable.iterator(PagedIterable.java:23) ~[stash-util-3.6.0.jar:na]
        at com.google.common.collect.Iterables$8.iterator(Iterables.java:709) ~[guava-11.0.2-atlassian-01.jar:na]
        at com.google.common.collect.Iterables.iterators(Iterables.java:499) ~[guava-11.0.2-atlassian-01.jar:na]
        at com.google.common.collect.Iterables.access$100(Iterables.java:60) ~[guava-11.0.2-atlassian-01.jar:na]
        at com.google.common.collect.Iterables$2.iterator(Iterables.java:489) ~[guava-11.0.2-atlassian-01.jar:na]
        at com.google.common.collect.Iterables$3.next(Iterables.java:508) ~[guava-11.0.2-atlassian-01.jar:na]
        at com.google.common.collect.Iterables$3.next(Iterables.java:501) ~[guava-11.0.2-atlassian-01.jar:na]
        at com.google.common.collect.Iterators$5.hasNext(Iterators.java:544) ~[guava-11.0.2-atlassian-01.jar:na]
        at com.google.common.collect.Multimaps.index(Multimaps.java:1853) ~[guava-11.0.2-atlassian-01.jar:na]
        at com.google.common.collect.Multimaps.index(Multimaps.java:1784) ~[guava-11.0.2-atlassian-01.jar:na]
        at org.christiangalsterer.stash.filehooks.plugin.hook.FileSizeHook.zipWithSize(FileSizeHook.java:112) ~[na:na]
        at org.christiangalsterer.stash.filehooks.plugin.hook.FileSizeHook.getChanges(FileSizeHook.java:107) ~[na:na]
        at org.christiangalsterer.stash.filehooks.plugin.hook.FileSizeHook.onReceive(FileSizeHook.java:81) ~[na:na]
        at com.atlassian.stash.internal.hook.repository.PreReceiveRepositoryHookAdapter$1.visit(PreReceiveRepositoryHookAdapter.java:38) [stash-service-impl-3.6.0.jar:na]
        at com.atlassian.stash.internal.hook.repository.PreReceiveRepositoryHookAdapter$1.visit(PreReceiveRepositoryHookAdapter.java:33) [stash-service-impl-3.6.0.jar:na]
```

which I imagine is because of the `00000` that you get as a ref.

**Comment by christiangalsterer** [comment]
Identified the problem and have a local fix available. Will need to test and hope that I can make a release in the next days.

**Comment by itay** [comment]
Thanks @christiangalsterer - let us know once it is pushed and we will install an updated version.

**Comment by christiangalsterer** [comment]
Fixed with new release 1.1.0.
